core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala

```diff
@@ -234,11 +234,15 @@ case class GitHubDirective(currentPath: String, variables: Map[String, String])
  *
  * Extracts snippets from source files into verbatim blocks.
  */
-case class SnipDirective(page: Page) extends LeafBlockDirective("snip") {
+case class SnipDirective(page: Page, variables: Map[String, String]) extends LeafBlockDirective("snip") {
+  private lazy val snippetBase =
+    new File(variables.getOrElse("snippet.base_dir", sys.error("Property `snippet.base_dir` is not defined")))
```

Comment:
Won't this explode in the cases where you don't define it? (Same for image.base_url)